### ISSUE TITLE
ansible_paramiko_user is ignored

### ISSUE THREAD

**Issue by jdashton** [ansible/ansible#52103]
##### SUMMARY
The docs for [paramiko_ssh](https://docs.ansible.com/ansible/latest/plugins/connection/paramiko_ssh.html) state that `remote_user` can be set using the var: `ansible_paramiko_user`or environment variable `ANSIBLE_PARAMIKO_REMOTE_USER`. However, these are ignored, even when `ansible_connection=paramiko` is present.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`_play_context` setup for connection plugins (probably in `playbook/play_context.py`.

##### ANSIBLE VERSION
```
$ ansible --version
ansible 2.7.6
  config file = None
  configured module search path = ['/Users/jdashton/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.7.6/libexec/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.2 (default, Feb 12 2019, 08:15:36) [Clang 10.0.0 (clang-1000.11.45.5)]
$
```

##### CONFIGURATION
```
$ ansible-config dump --only-changed
$
```

##### OS / ENVIRONMENT
macOS Mojave Version 10.14.4 Beta (18E184e)

##### DEPENDENCY PROBLEMS
```paste below

```

##### STEPS TO REPRODUCE
1. In an empty directory, cat the following to `hosts-test`:
```
my-host  ansible_host=192.168.162.101 ansible_paramiko_user=my_user ansible_connection=paramiko
```
2. Run this command:
```
ansible all -i hosts-test -m ping -vvv
```

##### EXPECTED RESULTS
Output should include this line with the `my_user` remote user:
```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: my_user on PORT 22 TO 192.168.162.101
```

##### ACTUAL RESULTS
Output includes this line with `None` as the remote user:
```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: None on PORT 22 TO 192.168.162.101
```

**Comment by jdashton** [comment]
Also true for the `ANSIBLE_PARAMIKO_REMOTE_USER` environment variable.
##### STEPS TO REPRODUCE
Using the same `hosts-test` file above, run this command:
```
ANSIBLE_PARAMIKO_REMOTE_USER=my_user2 ansible all -i hosts-test -m ping -vvv
```

##### EXPECTED RESULTS
Output should include this line with the `my_user2` remote user:

```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: my_user2 on PORT 22 TO 192.168.162.101
```
##### ACTUAL RESULTS
Output includes this line with `None` as the remote user:

```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: None on PORT 22 TO 192.168.162.101
```

**Comment by jdashton** [comment]
Also true for the `remote_user` in the `[paramiko_connection]` section of the ini file.

##### STEPS TO REPRODUCE
1. Cat the following to `ansible.cfg` in the current directory:
```
[paramiko_connection]
remote_user = my_user3
```

2. Using the same `hosts-test` file above, run this command:
```
ansible all -i hosts-test -m ping -vvv
```

##### EXPECTED RESULTS
Output should include this line with the `my_user3` remote user:
```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: my_user3 on PORT 22 TO 192.168.162.101
```

##### ACTUAL RESULTS
Output includes this line with `None` as the remote user:
```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: None on PORT 22 TO 192.168.162.101
```

**Comment by sivel** [comment]
This is very much true. I would say that we are in a transitional phase at the moment.  These new config values that are listed in the `DOCUMENTATION` of the `paramiko_ssh.py` file are not the source of truth, and are thus ignored.

I cannot see us making them actually work right now, and making them work will depend on transitioning more away from using `PlayContext`.

I'm leaning towards this being a documentation error, instead of a bug.

**Comment by bcoca** [comment]
@jdashton can you confirm PR above fixes the issue for you?

**Comment by jdashton** [comment]
Confirmed, this resolves all three of the issues I listed. (I had to modify the descriptions above because I had failed to anticipate the order of precedence.) This would be lovely to have in production.

**Comment by bcoca** [comment]
had to revert, some unforseen consequences in related code

**Comment by samdoran** [comment]
Waiting to hear back from the networking team to make sure this doesn't cause issues for them.

**Comment by sivel** [comment]
This has been resolved by a1bff416edf9b9c8bd5c3b002277eed5b5323953
